nifi-nar-bundles/nifi-spark-bundle/nifi-livy-controller-service/src/main/java/org/apache/nifi/controller/livy/LivySessionController.java

```diff
@@ -551,4 +561,11 @@ private SSLContext getSslSocketFactory(SSLContextService sslService)
         return sslContext;
     }
 
+    private void checkSessionManagerError() throws IOException {
+        Exception exception = sessionManagerError;
+        if (exception != null) {
+            throw new IOException(exception);
```

Comment:
I'm thinking this should be a custom exception type, probably thrown from the manageSession() thread itself and just propagated to the client via this method, in order to be able to tell the difference between an IOException that occurred from the mgmt thread vs an IOException that occurred from the operation the client is trying to perform. Thoughts?

Reply:
good idea @mattyb149, added a new SessionManagerException